File: java/src/main/java/deltix/luminary/UUIDDef.java
package deltix.luminary;

import java.util.ArrayList;
import java.util.List;

public class UUIDDef implements TypeDef<TypeUUID>, ConstantScope {
    private List<ConstantDef> definedConstants;

    private UUIDDef() {
        definedConstants = new ArrayList<>();
        definedConstants.add(new ConstantDef(this, "MIN_VALUE", TypeUUID.INSTANCE, new LiteralUUID(LiteralUUID.MIN_VALUE)));
        definedConstants.add(new ConstantDef(this, "MAX_VALUE", TypeUUID.INSTANCE, new LiteralUUID(LiteralUUID.MAX_VALUE)));
    }

    public static final UUIDDef INSTANCE = new UUIDDef();

    @Override
    public String getName() {
        return "UUID";
    }

    @Override
    public String getFullName() {
        return getName();
    }

    @Override
    public List<ConstantDef> getDefinedConstants() {
        return definedConstants;
    }

    @Override
    public TypeUUID getType() {
        return TypeUUID.INSTANCE;
    }

    @Override
    public FileDef getFile() {
        return null;
    }

    @Override
    public TypeScope getParent() {
        return null;
    }
}

File: csharp/main/UUIDDef.cs
using System;
using System.Collections.Generic;

namespace Deltix.Luminary
{
	// ReSharper disable once InconsistentNaming
	public class UUIDDef : IConstantScope, ITypeDef<TypeUUID>
	{
		private UUIDDef()
		{
			DefinedConstants.Add(new ConstantDef(this, "MIN_VALUE", TypeTime.Instance, new LiteralUUID(LiteralUUID.MinValue)));
			DefinedConstants.Add(new ConstantDef(this, "MAX_VALUE", TypeTime.Instance, new LiteralUUID(LiteralUUID.MaxValue)));
		}

		public static UUIDDef Instance = new UUIDDef();

		public String Name => "UUID";

		public String FullName => Name;

		public List<ConstantDef> DefinedConstants { get; } = new List<ConstantDef>();

		public TypeUUID Type => TypeUUID.Instance;

		public FileDef File => null;

		public ITypeScope Parent => null;
	}
}
